interfaces/tests/uuid_int/uuid_int.c

```diff
@@ -219,27 +219,21 @@ TEST_FUNCTION(PRI_UUID_T_compiles)
 
 static void UUID_T_VALUES_OR_NULL_helper(UUID_T u)
 {
-    LogInfo("u=%" PRI_UUID_T "", UUID_T_VALUES_OR_NULL(u));
+    LogInfo("u=%" PRI_UUID_T "", UUID_T_VALUES(u));
 }
 
 TEST_FUNCTION(UUID_T_VALUES_OR_NULL_compiles)
 {
     ///arrange
     UUID_T u;
-    ASSERT_ARE_EQUAL(int, 0, uuid_produce(u));
+    ASSERT_ARE_EQUAL(int, 0, uuid_produce(&u));
 
     ///act 1 (well - it compiles, doesn't crash at runtime)
-    UUID_T_VALUES_OR_NULL_helper(NULL); /*needs a helper to decay array into pointer*/
+    LogInfo("NULL UUID_T*=%" PRI_UUID_T "", PUUID_T_VALUES_OR_NULL((UUID_T*)NULL));
```

Comment:
>(UUID_T*) [](http://example.com/codeflow?start=67&length=9)

Why do we need a cast?

Reply:
error C2223: left of '->bytes' must point to struct/union